src/graph.c

```diff
@@ -243,8 +243,9 @@ int graph_refine_edge (graph_t *G, unsigned int va, unsigned int vb,
   /* FIXME: graph construction from peptide is still a bit ill-defined. */
 
   /* check if an exact edge already exists. */
-  if (graph_has_edge(G, va, vb) == VALUE_TYPE_SCALAR)
-    return 1;
+  /* Therese Malliavin */
+  /* if (graph_has_edge(G, va, vb) == VALUE_TYPE_SCALAR)
+    return 1; */
```

Comment:
I'm curious, what is the justification for removing this line? The idea was that edges with exact weights cannot be refined further, but this change allows exact weights to be modified.